[Improvement-16443][Workflow Timing] Improvement add operator

sdhzwc opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler, we are happy that you want to help us improve DolphinScheduler! -->
close #16443
The effect is as follows：
![recording](https://github.com/user-attachments/assets/0cb70a94-8ac4-4165-970b-a4f84cd5ce82)


## Purpose of the pull request

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
- *Add maven-checkstyle-plugin to root pom.xml*
-->

## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
- *Added dolphinscheduler-dao tests for end-to-end.*
- *Added CronUtilsTest to verify the change.*
- *Manually verified the change by testing locally.* -->

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

Commit:
[Improvement-16443][Workflow Timing] Improvement add operator

Commit:
[Improvement-16443][Workflow Timing] Improvement add operator

SbloodyS commented on dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Schedule.java:
```diff
@@ -108,10 +108,21 @@ public class Schedule {
-     * created user name
+     * operator user id
+     */
+    private Integer operator;
```
It's better to set an explicit name like `modifyUserId`.

sdhzwc replied:
> It's better to set an explicit name like `modifyUserId`.
This one was written with reference to the previous naming, and I thought it would be better to keep it uniform, such as the one below:
![image](https://github.com/user-attachments/assets/a9098224-20fe-4bfd-8b6e-0885f7f83281)

SbloodyS replied:
These are problems left over from history and will be reconstructed in the future

sdhzwc replied:
> These are problems left over from history and will be reconstructed in the future

Modifications completed

Commit:
operator changed to modify_user_id

Commit:
operator changed to modify_user_id

SbloodyS committed:
Merge branch 'dev' into Improvement-16443

SbloodyS commented on dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql:
```diff
@@ -125,3 +125,5 @@ d//
 CALL drop_column_t_ds_alert_plugin_instance;
 DROP PROCEDURE drop_column_t_ds_alert_plugin_instance;
+
+ALTER TABLE t_ds_schedules ADD  `modify_user_id` int(11) DEFAULT NULL COMMENT 'modify user id';
```
Please leave a blank line at the end of each file.

sdhzwc replied:
> Please leave a blank line at the end of each file.

done

SbloodyS commented on dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_ddl.sql:
```diff
@@ -123,3 +123,5 @@ d//
 select drop_column_t_ds_alert_plugin_instance();
 DROP FUNCTION IF EXISTS drop_column_t_ds_alert_plugin_instance();
+
+ALTER TABLE t_ds_schedules ADD COLUMN IF NOT EXISTS modify_user_id int;
```
Same here.

Commit:
add a blank line

Commit:
Merge remote-tracking branch 'origin/Improvement-16443' into Improvement-16443

SbloodyS reviewed (dismissed):
LGTM

luoyize committed:
[Fix][pom] Fix jacoco-maven-plugin error config . (#16463)

Dyqer committed:
[Fix-16382] Fix the bug of async master task casthread pool invocations ramp-up

ruanwenjun reviewed (commented):
It's better to set `modify_user` as create_user when upgrade.

sdhzwc commented:
> It's better to set `modify_user` as create_user when upgrade.

What needs to be adjusted? Could you specify?

SbloodyS committed:
[DSIP-62][Http Alert Plugin] Refactor http alert plugin (#16484)

* add dsip_62

* fix sonar

* fix spotless

ruanwenjun commented:
> > It's better to set `modify_user` as create_user when upgrade.
> 
> What needs to be adjusted? Could you specify?

It's not a good idea to make `modify_user_id` DEFAULT NULL, this will make confusion, why the create user is not null, but the modify_user_id can be null, no one can explain this. If the modify_user_id is null is this a bug?

To avoid this kind problem, you need to inject a value for the new column, obviously you should use create_user to inject.

ruanwenjun commented:
Additional, it's better to create a DSIP to unify the logic, we should add modify user to all table which have create user. Otherwise, this look like strange.

sdhzwc commented:
> > > It's better to set `modify_user` as create_user when upgrade.
> > 
> > 
> > What needs to be adjusted? Could you specify?
> 
> It's not a good idea to make `modify_user_id` DEFAULT NULL, this will make confusion, why the create user is not null, but the modify_user_id can be null, no one can explain this. If the modify_user_id is null is this a bug?
> 
> To avoid this kind problem, you need to inject a value for the new column, obviously you should use create_user to inject.

The reason is very simple. When the timer is created, it is definite. The creator is the logged-in person. At this time, there is no modification operation, so there is no need to specify the updater. I think it makes sense. If the updater is added, I think there will be ambiguity because there is no update operation yet.

sdhzwc commented:
> nge.

Workflow Definition has Create User and Modify User.
Workflow Instance and Task Instance have Executor.
The following two are not clear and need to add corresponding Create User and Modify User. This can be handled as a DSIP: Workflow Timing and Task Definition.

> Additional, it's better to create a DSIP to unify the logic, we should add modify user to all table which have create user. Otherwise, this look like strange.

Workflow Definition has Create User and Modify User.
Workflow Instance and Task Instance have Executor.
The following two are not clear and need to add corresponding Create User and Modify User. This can be handled as a DSIP: Workflow Timing and Task Definition.

sdhzwc commented:
> Additional, it's better to create a DSIP to unify the logic, we should add modify user to all table which have create user. Otherwise, this look like strange.

Workflow Definition has Create User and Modify User.
Workflow Instance and Task Instance have Executor.
The following two are not clear and need to add corresponding Create User and Modify User. This can be handled as a DSIP: Workflow Timing and Task Definition.

ruanwenjun commented:
> > > > It's better to set `modify_user` as create_user when upgrade.
> > > 
> > > 
> > > What needs to be adjusted? Could you specify?
> > 
> > 
> > It's not a good idea to make `modify_user_id` DEFAULT NULL, this will make confusion, why the create user is not null, but the modify_user_id can be null, no one can explain this. If the modify_user_id is null is this a bug?
> > To avoid this kind problem, you need to inject a value for the new column, obviously you should use create_user to inject.
> 
> The reason is very simple. When the timer is created, it is definite. The creator is the logged-in person. At this time, there is no modification operation, so there is no need to specify the updater. I think it makes sense. If the updater is added, I think there will be ambiguity because there is no update operation yet.

This is not a common practice, please reference the last_modify_user/last_modify_time in aws/linux.

Commit:
insert add modify_user_id

sdhzwc commented:
> > > > > It's better to set `modify_user` as create_user when upgrade.
> > > > 
> > > > 
> > > > What needs to be adjusted? Could you specify?
> > > 
> > > 
> > > It's not a good idea to make `modify_user_id` DEFAULT NULL, this will make confusion, why the create user is not null, but the modify_user_id can be null, no one can explain this. If the modify_user_id is null is this a bug?
> > > To avoid this kind problem, you need to inject a value for the new column, obviously you should use create_user to inject.
> > 
> > 
> > The reason is very simple. When the timer is created, it is definite. The creator is the logged-in person. At this time, there is no modification operation, so there is no need to specify the updater. I think it makes sense. If the updater is added, I think there will be ambiguity because there is no update operation yet.
> 
> This is not a common practice, please reference the last_modify_user/last_modify_time in aws/linux.

It has been changed. The value has been added when inserting.

topsli committed:
[Fix-16475][UI] Can't SET Timing successful in Workflow Definition grid (#16476)

saowu committed:
[Fix-16492][alert] FeiShuSender Exception.getMessage is null (#16498)

* Update FeiShuSender.java

The Exception.getMessage present here is null, and the problem has occurred

* Update dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/FeiShuSender.java

Co-authored-by: xiangzihao <zihaoxiang@apache.org>

---------

Co-authored-by: xiangzihao <zihaoxiang@apache.org>

BruceWong96 committed:
[Doc-16490] Changed some notes about python gateway service (#16493)

shouwangyw committed:
[Improvement-16487][Master] Improve the dependent result for workflow dependency tasks (#16488)

lyyprean committed:
[Improvement-16501][dolphinscheduler-api]typo in Status.java（#16501） (#16504)

cncws committed:
[Fix-15961] [UI] Rename pythonCommand to pythonLauncher in pytorch task. (#16495)

ruanwenjun committed:
[DSIP-61][Master] Refactor thread pool and  state event orchestration in master (#16327)

ruanwenjun committed:
[DSIP-67] Use command to trigger workflow instance rather generate workflow instance (#16523)

SbloodyS committed:
[Improvement-16516] Uniformly name process in backend code as workflow (#16524)

pegasas committed:
[Improvement][API] Fix typo on api workflow backfill (#16538)

wangxj3 committed:
[Doc-16535] remove doc of extra ssh task node type (#16536)

SbloodyS committed:
remove staleness code owner (#16547)

ruanwenjun committed:
[Improve] Move recover/repeat running to IWorkflowControlClient (#16545)

SbloodyS committed:
[DSIP-70] Deprecated upgrade code of 1.x and 2.x in 3.3.0-release (#16551)

ruanwenjun committed:
[Chore] Remove unsupported feature in README.md (#16549)

* [Chore] Remove unsupported feature in README.md

* Update README_zh_CN.md

Co-authored-by: xiangzihao <zihaoxiang@apache.org>

---------

Co-authored-by: xiangzihao <zihaoxiang@apache.org>
Co-authored-by: xiangzihao <460888207@qq.com>

SbloodyS committed:
[Chore] Improve release doc (#16553)

* improve release doc

Commit:
Merge remote-tracking branch 'origin/Improvement-16443' into Improvement-16443

# Conflicts:
#	dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java
#	dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql
#	dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_ddl.sql
